receiver/otlpreceiver/otlphttp.go

```diff
@@ -28,3 +34,35 @@ type xProtobufMarshaler struct {
 func (*xProtobufMarshaler) ContentType() string {
 	return "application/x-protobuf"
 }
+
+// oTLPErrorHandler encodes the HTTP error message inside a rpc.Status message as required
+// by the OTLP protocol.
+func oTLPErrorHandler(w http.ResponseWriter, r *http.Request, err error, statusCode int) {
+	var msg []byte
+	var s *status.Status
+	fallbackMsg := []byte(`{"code": 13, "message": "failed to marshal error message"}`)
+
+	switch statusCode {
+	case http.StatusBadRequest:
+		s = status.New(codes.InvalidArgument, err.Error())
+	default:
+		s = status.New(codes.Internal, err.Error())
+	}
+
+	contentType := r.Header.Get("Content-Type")
+	switch contentType {
+	case "application/json":
+		msg, err = json.Marshal(s.Proto())
```

Comment:
Can we use [jsonpb](https://godoc.org/github.com/gogo/protobuf/jsonpb) instead of json?
There is likely no difference for `status.Status` but just to be safe that it correctly uses Protobuf JSON mapping.